### PR TITLE
Move JWS validation from samples to source

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: tests/ut/mbedtls/mbedtls_freertos_port.c [hunk 1]
```diff
@@ -0,0 +1,239 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+/**
+ * @file mbedtls_freertos_port.c
```
Comment (CIPop) [inline tests/ut/mbedtls/mbedtls_freertos_port.c:5]: Nit: we should add that this is a test-only implementation.

File: tests/ut/azure_iot_jws_mbedtls_ut.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+static int setup( void ** state )
+{
+    memset( ucScratchBuffer, 0, sizeof( ucScratchBuffer ) );
+    return 0;
+}
+
+static void testAzureIoTJWS_ManifestAuthenticate_Success( void ** ppvState )
+{
+    assert_int_equal( AzureIoTJWS_ManifestAuthenticate( ucValidManifest, strlen( ucValidManifest ),
```
Comment (CIPop) [inline tests/ut/azure_iot_jws_mbedtls_ut.c:146]: I think the expected value should be the first parameter. 
Some UT frameworks will differentiate and the error message when the test fails may be confusing. Even if cmocka doesn't, please take it as a nit (for code readability).

**Review by CIPop — APPROVED** [review]
Left a couple of nitpicks.
Thanks @danewalton !

File: tests/ut/azure_iot_jws_mbedtls_ut.c (hunk 2 above)
Comment (danewalton) [reply tests/ut/azure_iot_jws_mbedtls_ut.c:146]: I hear you. Unfortunately most of the tests (if not all) are this way:

image

**Pull request merged**